--- nss_updatedb-10/configure.in.link_db~	2012-05-30 21:32:19.130866901 +0200
+++ nss_updatedb-10/configure.in	2012-05-30 21:35:47.716259225 +0200
@@ -18,7 +18,7 @@ AC_CHECK_HEADERS(db3/db.h)
 AC_CHECK_HEADERS(db4/db.h)
 AC_CHECK_HEADERS(db_nss/db.h)
 
-AC_CHECK_LIB(db_nss, main,[LIBS="-ldb_nss $LIBS" found_db_lib=yes],,$LIBS)
+AC_CHECK_LIB(db_nss, main,[LIBS="-ldb_nss $LIBS" found_db_nss_lib=yes],,$LIBS)
 if test -z "$found_db_lib"; then
 	AC_CHECK_LIB(db, main,[LIBS="-ldb $LIBS" found_db_lib=yes],,$LIBS)
 fi
